### PR TITLE
Multi-Form Goal total now includes renewals

### LINKED ISSUE #5358: Multi-Form Goal total should include recurring donation renewals
Currently, the direct query only accounts for payments with a status of `publish`, but recurring donation renewals use the status of `give_subscription`.

## Acceptance Criteria

- [ ] Renewal payments are included in the goal total.

### PR DESCRIPTION
**Pull request by kjohnson**
Closes #5358 

## Description

Includes revenue items for payments with a status of `publish` and `give_subscription`.

```sql
            SELECT
                sum( revenue.amount ) as total,
                count( payment.ID ) as count
            FROM wp_posts as payment
                JOIN wp_give_revenue as revenue
                    ON revenue.donation_id = payment.ID
            WHERE
                payment.post_type = 'give_payment'
                AND
                payment.post_status IN ( 'publish', 'give_subscription' )
```

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [x] Changelog updated
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

## Testing Instructions

- Create a recurring donation.
- Manually trigger a renewal.
- Goal total should include both the donation and the renewal

### CODE REVIEW

**Comment by kjohnson** [comment]
@henryholtgeerts  This raises the question of if the `status` check in the direct query is unnecessary, or even if it is a hindrance.

@ravinderk how does the revenue table account for payment status?

**Comment by henryholtgeerts** [comment]
@kjohnson This is an interesting point - I'm also curious how post status is handled with the revenue table. Historically, post status checks have been critical because a donation could be marked as any number of non-completed status (think refunded, failed, etc), but if the revenue table is limited solely to completed payments or renewals, than it makes sense to drop the status check.

**Comment by ravinderk** [comment]
@kjohnson Status check is necessary to query in the revenue table but remaining depends upon the requirement. The SQL query mentioned in pr description looks good to me.

File: src/MultiFormGoals/ProgressBar/Query.php [hunk 1]
```diff
@@ -25,29 +25,21 @@ public function __construct( $formIDs ) {
...
-                    ON meta.donation_id = payment.ID
+                JOIN wp_give_revenue as revenue
+                    ON revenue.donation_id = payment.ID
             WHERE
                 payment.post_type = 'give_payment'
                 AND
-                meta.meta_key = '_give_payment_total'
-                AND
-                payment.post_status = 'publish'
+                payment.post_status IN ( 'publish', 'give_subscription' )
```
Comment (ravinderk) [inline src/MultiFormGoals/ProgressBar/Query.php:36]: @kjohnson I think we can add a method in `Revenue` repository class to add `post_type` and `post_type` in the SQL query. I can see that will be helpful for most of the SQL queries in the revenue table.

**Comment by kjohnson** [comment]
@henryholtgeerts @JasonTheAdams still needing a review on this one.

**Review by JasonTheAdams — APPROVED** [review]
This looks good to me. At some point it would be great to have some kind of Revenue query builder, but we don't right now so this is good. 👍

**Comment by JasonTheAdams** [comment]
@kjohnson The only thing I'm wondering here is if the result should be canceled. This shows up on the front-end, right? If so, that would mean the query would currently happen for every request?

**Comment by kjohnson** [comment]
> @kjohnson The only thing I'm wondering here is if the result should be canceled. This shows up on the front-end, right? If so, that would mean the query would currently happen for every request?

@JasonTheAdams do you mean to say `cached`?

**Comment by JasonTheAdams** [comment]
I sure did. 😆

**Comment by kjohnson** [comment]
@JasonTheAdams ok, that makes a lot more sense :) So, yeah. I can open a second PR for caching the Query, but this is specifically about including Renewal Payments.

**Pull request merged**